api/rosetta/data_balance_integration_test.go

```diff
@@ -90,69 +104,79 @@ func TestGetBalance(t *testing.T) {
 	db := setupDB(t)
 	api := setupAPI(t, db)
 
+	const (
+		testAccount = "754aed9de6197641"
+	)
+
+	var (
+		// block where the account first appears
+		first = identifier.Block{
+			Index: 13,
+			Hash:  "af528bb047d6cd1400a326bb127d689607a096f5ccd81d8903dfebbac26afb23",
+		}
+
+		// a block mid-chain
+		second = identifier.Block{
+			Index: 50,
+			Hash:  "d99888d47dc326fed91087796865316ac71863616f38fa0f735bf1dfab1dc1df",
+		}
+
+		// last indexed block
+		last = identifier.Block{
+			Index: 425,
+			Hash:  "594d59b2e61bb18b149ffaac2b27b0efe1854f6795cd3bb96a443c3676d78683",
+		}
+	)
+
 	tests := []struct {
 		name string
 
 		request rosetta.BalanceRequest
 
 		wantStatusCode int
-		wantBalance    string
-		wantHandlerErr assert.ErrorAssertionFunc
+		wantBalances   []string
+		validateBlock  blockIDValidationFn
 	}{
 		{
-			name:           "valid balance request - first occurrence of the account",
-			request:        balanceRequest("754aed9de6197641", 13, "af528bb047d6cd1400a326bb127d689607a096f5ccd81d8903dfebbac26afb23"),
-			wantBalance:    "10000100000",
-			wantStatusCode: http.StatusOK,
-			wantHandlerErr: assert.NoError,
+			name:          "first occurrence of the account",
+			request:       balanceRequest(testAccount, first),
```

Comment:
I commented on this before, but `BalanceRequest` is a struct. Functions are much better when they use action words (verbs), so in this case: `createBalanceRequest(testAccount, firstBlock)`, or maybe a bit shorter `requestBalance(testAccount, firstBlock)`.

Reply:
Yeah, my first version was `getBalanceRequest`. Renamed to `requestBalance`